File: golfBackend/models/reservation.py

from db import db
from datetime import datetime

class ReservationModel(db.Model):
    __tablename__ = 'reservations'

    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)

    date = db.Column(db.Date, nullable=False)
    start_time = db.Column(db.Time, nullable=False)
    end_time = db.Column(db.Time, nullable=False)
    duration_minutes = db.Column(db.Integer, nullable=False)

    created_at = db.Column(db.DateTime, default=datetime.now)

    user = db.relationship("UserModel", back_populates="reservations")
    reservation_items = db.relationship("ReservationItemModel", back_populates="reservation", cascade="all, delete") 

File: golfFrontend/src/types/reservation.ts
export type Service = {
  name: string;
  category?: string;
};

export type ReservationItem = {
  quantity: number;
  service: Service;
};

export type Reservation = {
  id: number;
  date: string;
  start_time: string;
  end_time: string;
  reservation_items: ReservationItem[];
};
